### PR TITLE
[WIP] MATE 1.20

### PR DESCRIPTION
**Pull request by Mno-hime**
Requires newer GTK3.

So far making MATE 1.20 as a one big PR dump for easier packaging, will
chop to pieces as needed.

Problems:
* Pluma's text area flashes (https://github.com/mate-desktop/pluma/issues/282).
* Nimbus theme needs to be fixed.

### CODE REVIEW

**Comment by pyhalov** [comment]
@Mno-hime There's one big issue with this - GTK3  update requires nimbus update/rewrite...

**Comment by Mno-hime** [comment]
@pyhalov Yes, I am aware of that, thought I did not tackle that yet. Even thinking why we stick with an ancient theme (and icons), which is guaranteed to break on MATE/GTK+ update? I run MATE 1.20 so I stick with "TradationalOk" theme and I find it a very pleasant experience.

**Comment by pyhalov** [comment]
When I switch to it, I immediately see some artifacts, like no spaces in mate-panel  between menus, no menu highlighting on selection and so on. It just feels incomplete. If you find another theme, which looks nice and is supported, we can look at switching default one.

**Comment by Mno-hime** [comment]
Might be your local problem of sorts? I have very recent installation of OI updated with MATE 1.20. This is how it looks like with TraditionalOk theme (and MATE icon set, default for that theme):
traditionalok
I don't see any glitches here.

**Comment by Mno-hime** [comment]
Updated to the latest MATE 1.20 packages. Next I'll have a look at the themes.

**Comment by Mno-hime** [comment]
@pyhalov How does one approach themes broken by GTK update? I see errors like this `** (mate-panel:2623): WARNING **: 10:45:14.452: unsupported value of 'background-image' in GTK+ theme (such as '-gtk-gradient')` when changing theme to Nimbus. Looks like a real error? Is there other way to fix it than manual intervention.

**Comment by Mno-hime** [comment]
With help of pkgsrc's vte3 patches I updated vte-291 component to the latest release and MATE terminal 1.20.1 now works as expected.

What's left:

- [ ] Nimbus
- [x] Update to the latest 1.20 packages
- [ ] Blinking Pluma (at least for me)

**Comment by pyhalov** [comment]
1) Have you looked at nimbus? 
2) What do you think about merging gtk3 prereqs to facilitate gtk testing?

**Comment by pyhalov** [comment]
While building libepoxy I get warning:
```config.h:103:0: warning: "_FILE_OFFSET_BITS" redefined```
It could be dangerous (having sources built with different _FILE_OFFSET_BITS). Perhaps, we have to add 
```CONFIGURE_ENV.32 += CPPFLAGS="$(CPP_LARGEFILES.32)"``` to libepoxy Makefile

**Comment by Mno-hime** [comment]
ad 1: Saw couple of github repos which claimed to be GTK3 Nimbus wasn't persuaded by any of them.
ad 2: Will prepare PR later today.
ad epoxy: Will do.

**Comment by pyhalov** [comment]
As far as I know, our nimbus has most of the GTK fixes, other seem to be dead.

**Comment by Mno-hime** [comment]
pango & libepoxy with configure fix: https://github.com/OpenIndiana/oi-userland/pull/4473

File: components/desktop/mate/mate-panel/mate-panel.p5m [hunk 1]
```diff
@@ -22,10 +23,6 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
 
 license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
 
-file files/bottom-panel-image.png path=usr/share/mate-panel/pixmaps/bottom-panel-image.png
```
Comment (pyhalov) [inline components/desktop/mate/mate-panel/mate-panel.p5m]: Please, preserve these files.

**Comment by Mno-hime** [comment]
Started chopping this to more digestible PRs.

**Comment by Mno-hime** [comment]
**Prerequisites**

1. GTK+3 3.24.1 update: https://github.com/OpenIndiana/oi-userland/pull/4507
2. Nimbus theme update: WIP by @pyhalov & Goeff

**MATE 1.20 steps**
1. https://github.com/OpenIndiana/oi-userland/pull/4508
2. https://github.com/OpenIndiana/oi-userland/pull/4509
3. https://github.com/OpenIndiana/oi-userland/pull/4510
4. https://github.com/OpenIndiana/oi-userland/pull/4511
5. https://github.com/OpenIndiana/oi-userland/pull/4512

So, closing this PR, additional fixes should happen in those PRs above.

**Pull request closed without merging**